fe2-android/app/src/test/java/com/github/dedis/popstellar/model/network/method/message/data/socialmedia/DeleteChirpTest.java

```diff
@@ -0,0 +1,65 @@
+package com.github.dedis.popstellar.model.network.method.message.data.socialmedia;
+
+import static com.github.dedis.popstellar.Base64DataUtils.generateMessageID;
+import static com.github.dedis.popstellar.Base64DataUtils.generateMessageIDOtherThan;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertThrows;
+
+import com.github.dedis.popstellar.model.network.JsonTestUtils;
+import com.github.dedis.popstellar.model.network.method.message.data.Action;
+import com.github.dedis.popstellar.model.network.method.message.data.Objects;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.google.gson.JsonParseException;
+
+import org.junit.Test;
+
+public class DeleteChirpTest {
+
+  private static final MessageID CHIRP_ID = generateMessageID();
+  private static final long TIMESTAMP = 1631280815;
+
+  private static final DeleteChirp DELETE_CHIRP = new DeleteChirp(CHIRP_ID, TIMESTAMP);
+
+  @Test
+  public void getObjectTest() {
+    assertEquals(Objects.CHIRP.getObject(), DELETE_CHIRP.getObject());
+  }
+
+  @Test
+  public void getActionTest() {
+    assertEquals(Action.DELETE.getAction(), DELETE_CHIRP.getAction());
+  }
+
+  @Test
+  public void getChirpIdTest() {
+    assertEquals(CHIRP_ID, DELETE_CHIRP.getChirpId());
+  }
+
+  @Test
+  public void getTimestampTest() {
+    assertEquals(TIMESTAMP, DELETE_CHIRP.getTimestamp());
+  }
+
+  @Test
+  public void equalsTest() {
+    assertEquals(DELETE_CHIRP, new DeleteChirp(CHIRP_ID, TIMESTAMP));
+
+    assertNotEquals(DELETE_CHIRP, new DeleteChirp(generateMessageIDOtherThan(CHIRP_ID), TIMESTAMP));
+    assertNotEquals(DELETE_CHIRP, new DeleteChirp(CHIRP_ID, TIMESTAMP + 1));
+  }
+
+  @Test
+  public void jsonValidationTest() {
+    JsonTestUtils.testData(DELETE_CHIRP);
+
+    String pathDir = "protocol/examples/messageData/chirp_delete_publish/";
+    assertThrows(
+        JsonParseException.class,
+        () -> JsonTestUtils.parse(pathDir + "wrong_chirp_delete_publish_negative_time.json"));
+    assertThrows(
+        JsonParseException.class,
+        () ->
+            JsonTestUtils.parse(pathDir + "wrong_chirp_delete_publish_not_base_64_chirp_id.json"));
```

Comment:
Sorry I forgot to add comments in the JsonTestUtils, and there was a confusion on how to use it.
The JsonTestUtils.parse take a json String as argument
The JsonTestUtils.loadFile take a path and return a json String
In this case (and in the others jsonValidationTest from addChirp/etc), we need to load the file into a String before testing it.

And it seems that there is another problem when it's not a Base64 (An IllegalArgumentException is thrown by the Base64URLData but was not catched and rethrown with an JsonParseException)

Reply:
```Suggestion
String pathDir = "protocol/examples/messageData/chirp_delete_publish/";
assertThrows(
    JsonParseException.class,
    () ->
        JsonTestUtils.parse(
            JsonTestUtils.loadFile(pathDir + "wrong_chirp_delete_publish_negative_time.json")));
assertThrows(
    JsonParseException.class,
    () ->
        JsonTestUtils.parse(
            JsonTestUtils.loadFile(
                pathDir + "wrong_chirp_delete_publish_not_base_64_chirp_id.json")));
```

Reply:
For the JsonBase64DataSerializer.java problem, this should fix the problem normally (I didn't check in details) :
```
@Override
public T deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
    throws JsonParseException {
  try {
    return constructor.apply(json.getAsString());
  } catch (Exception e) {
    throw new JsonParseException(e);
  }
}
```

Reply:
Yes I think it should be in a dedicated PR as it's not related to mine. It's easier to track changes